tuxemon/core/states/world/worldstate.py

```diff
@@ -940,7 +940,7 @@ def change_map(self, map_name):
 
         # move to spawn position, if any
         for eo in self.game.events:
-            if eo.name.lower() == "player spawn":
+            if eo.name and eo.name.lower() == "player spawn":
```

Comment:
why would event objects not have a name?

Reply:
Usually it's not the case. I could see myself removing the name from random battle events, due to too many boxes in a small area causing the names to overlap in Tiled. Or when putting together a map and testing, as the mapper might forget about it and get a crash.

It's a very minor thing though; If you feel there's no need for the check this PR can be closed.